apps/web/app/routes/stacking/stacking.tsx

```diff
@@ -11,30 +12,36 @@ import { delay } from '@leather.io/utils';
 
 import type { Route } from './+types/stacking';
 
-export async function loader() {
+export async function loader({ params }: Route.LoaderArgs) {
   await delay(400);
-  return true;
-}
-
-// eslint-disable-next-line no-empty-pattern
-export function meta({}: Route.MetaArgs) {
-  return [{ title: 'Leather Earn - Stacking' }];
-}
 
-export default function EarnStackingRoute({ params }: Route.ComponentProps) {
   const isSlugValid = validatePoolSlug(params.slug);
 
   if (!isSlugValid) {
-    throw new Error(`Invalid pool slug: ${params.slug}`);
+    // eslint-disable-next-line @typescript-eslint/only-throw-error
+    throw new Response(`Invalid pool slug: ${params.slug}`, { status: 404 });
   }
 
   const poolSlug = params.slug as PoolSlug;
   const poolId = PoolSlugToIdMap[poolSlug];
   const poolName = PoolIdToDisplayNameMap[poolId];
 
+  return { poolName };
+}
+
+// eslint-disable-next-line no-empty-pattern
+export function meta({}: Route.MetaArgs) {
+  return [{ title: 'Leather Earn - Stacking' }];
+}
+
+export default function EarnStackingRoute({ loaderData }: Route.ComponentProps) {
   return (
     <StackingClientProvider>
-      <StackInPool poolName={poolName} />
+      <StackInPool poolName={loaderData.poolName} />
     </StackingClientProvider>
   );
 }
+
+export function ErrorBoundary({ error }: Route.ErrorBoundaryProps) {
+  return <ErrorContent error={error} />;
+}
```

Comment:
Why do we need this? If it just passes the error to his component, isn't that the same behaviour as the global one?